### ISSUE TITLE
Support Renkforce AOK-5055

### ISSUE THREAD

**Issue by pjmpessers** [nlrb/com.weather-sensors#20]
I had a Cresta DTX500 weather-station. The weather-station outside is for temperature, rain, windforce and wind-direction. After a day or something it picks up only signal and it was saying low on battery. When removing it got broken. So i bought the Renkforce AOK-5055 from conrad with the same outside weather-station. The signal of this new one can be picked up by the display of the Renkforce AOK-5055 and the display of the Cresta DTX-500. Both are 433mhz and both are now 1 meter of homey. Homey sees it as a Alecto V1, but has many CRC check errors or says the temperature are out of range. 

I installed the V1.2.2. My firmware is 1.5.6.rc-6

Can you help me?

[stats1513111761131.txt.zip](https://github.com/nlrb/com.weather-sensors/files/1553327/stats1513111761131.txt.zip)

On the picture on the left is the new Renkforce AOK-5055 and on the right the old Cresta DTX500. As you see the outside weather-station are the same.

![20171212_222304](https://user-images.githubusercontent.com/34476404/33909305-8bc9c54a-df8b-11e7-9b9c-9b283ab0c8ea.jpg)

**Comment by pjmpessers** [comment]
I also have one debug-file where you can see that it had a good readout once. Where the humidity is good, but i needed a offset of -32.8 for the temperature. Also is it possible to support the rain, windforce and wind-direction of this weather-station?

Thanks for your help.
[stats1513150440025.zip](https://github.com/nlrb/com.weather-sensors/files/1555294/stats1513150440025.zip)

**Comment by nlrb** [comment]
Thanks for the data. I will have a look at it. These messages are shorter (35 bytes) than what I would expect. Let me see if I can find a protocol description for them.

**Comment by nlrb** [comment]
BTW: the data of the Cresta DTX-500 - did you receive this with the Cresta or Alectov1 protocol?

**Comment by pjmpessers** [comment]
Thank you for your help.

Both. Cresta and Alectov1. The same thing is for the Renkforce AOK-5055. It also picks up Cresta and Alectov1.

Alectov1 gives me sometimes a good readout on both weatherstations. Just need to adjust the offset for temperature. But it only picks up one good readout a day. 
Cresta also gives a readout, but it is saying: "Invalid cresta message."

The renkforce does the same thing. Sometimes both also has a lacrosse TX2/3 readout by both. But never valid. Maybe it uses for every connected devices (Rain, Temperature, enz) different protocols?

I only can connect the one by devices with the ID 77 or TH77.

When in need for more information, let me know. ;-)

**Comment by nlrb** [comment]
What I need to know is with which protocol the data was received reliably without CRC errors.

**Comment by pjmpessers** [comment]
That came from the Alectov1 with one good readout. That was on the Cresta DTX500 and the Renkforce OAK-5055.

**Comment by nlrb** [comment]
So with Alectov1 you got a good read-out with correct temperature as well?

**Comment by pjmpessers** [comment]
No. With Alectov1 it is more random. Most of the signal is invalid. But sometimes it picks up one signal. And the temperature readout is all over the place. A moment ago i needed to change the offset to -32.6. And the latest readout it was closer to the real temperature and i needed a offset of -2.2. The weatherstation is in this moment not outside, but in house. Also many cresta: "Not a valid cresta message."
[stats1513180945047.zip](https://github.com/nlrb/com.weather-sensors/files/1556183/stats1513180945047.zip)

**Comment by pjmpessers** [comment]
On this moment it found a second Alectov1 tempature/ humitidy sensor on ID 154. See JSON file. ;-)
[stats1513185971913.zip](https://github.com/nlrb/com.weather-sensors/files/1556399/stats1513185971913.zip)

**Comment by nlrb** [comment]
Ok, so in your case often the leading zero is missing in the data.

Can you make a change in `app.json` and install the code via the CLI?

Try changing line 605 to `"minimalLength": 36,` and see if that works better.

**Comment by pjmpessers** [comment]
I made the change, but somehow i couldn't sideload it. When i gave the command NPM install -g athom-cli
i got a message npm WARN deprecated node-uuid@1.4.8: Use uuid module instead.
[com.weather-sensors-master.zip](https://github.com/nlrb/com.weather-sensors/files/1557113/com.weather-sensors-master.zip)

**Comment by nlrb** [comment]
That is just a warning, you can ignore that.

**Comment by pjmpessers** [comment]
Oke. But i don't see wireless weather sensor by settings apps.:-(

**Comment by nlrb** [comment]
Can you make a recording of the signal with https://developer.athom.com/tools/signals ?

**Comment by pjmpessers** [comment]
Not a problem. And thank you for your support.
[recording.zip](https://github.com/nlrb/com.weather-sensors/files/1557181/recording.zip)

**Comment by pjmpessers** [comment]
I Managed sideloading the app. I made the mistake of using athom project-install instead of athom project --install. On this moment (5 Minutes) know devices.
[stats1513202813265.zip](https://github.com/nlrb/com.weather-sensors/files/1557291/stats1513202813265.zip)

**Comment by pjmpessers** [comment]
This night it didn't find a device. But i see a invalid La crosse message. 
[stats1513227574570.zip](https://github.com/nlrb/com.weather-sensors/files/1557996/stats1513227574570.zip)

**Comment by pjmpessers** [comment]
I hear of people having problem with the new 1.5.6-rc6 and/or 1.5.6-rc7 in the 433mhz. For me it looks that it doesn't make a difference. The same problem was with v1.5.5. My AYCT-102 klikaanklikuit remote on 433mhz still works without a problem.
I now see that it also find a lacrosse TX2/3 sensor. Is it possible it uses more then one protocols combined together? For every sensor uses another protocol? Or is this just a false readout?

[stats1513436892146.zip](https://github.com/nlrb/com.weather-sensors/files/1564875/stats1513436892146.zip)

**Comment by nlrb** [comment]
I don't know yet which protocols your sensors use. I first want to have stable Homey firmware before I can investigate this further. There are definitely issues with 1.5.6.rc-6/7.

**Comment by pjmpessers** [comment]
I understand. I shall wait for a more stable firmware in the 433mhz band. 😉



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by pjmpessers** [comment]
Still no luck with the firmware 1.5.6.rc-8.  :-(

**Comment by pjmpessers** [comment]
Despite the new firmware update 1.5.6.RC-11 that would solve the problem with the wireless wheatherstations, the problem still excist. ;-)

**Comment by pjmpessers** [comment]
Also RC-13 doesn't solve it. Just a readout every +/- 3 hours or more.

**Comment by Rotzbua** [comment]
The device does not use a known protocol. 
You can lookup my analytic: https://github.com/Rotzbua/reverse_engineering_weatherstation_conrad
